source/extensions/transport_sockets/alts/config.cc

```diff
@@ -0,0 +1,144 @@
+#include "extensions/transport_sockets/alts/config.h"
+
+#include "envoy/config/transport_socket/alts/v2alpha/alts.pb.h"
+#include "envoy/config/transport_socket/alts/v2alpha/alts.pb.validate.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/transport_socket_config.h"
+
+#include "common/common/assert.h"
+#include "common/protobuf/protobuf.h"
+#include "common/protobuf/utility.h"
+
+#include "extensions/transport_sockets/alts/grpc_tsi.h"
+#include "extensions/transport_sockets/alts/tsi_socket.h"
+
+#include "absl/strings/str_join.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Alts {
+
+typedef CSmartPtr<grpc_alts_credentials_options, grpc_alts_credentials_options_destroy>
+    GrpcAltsCredentialsOptionsPtr;
+
+// Returns true if the peer's service account is found in peers, otherwise
+// returns false and fills out err with an error message.
+static bool doValidate(const tsi_peer& peer, const std::unordered_set<std::string>& peers,
```

Comment:
Nit: prefer using anonymous namespace in `.,cc` to `static` functions in Envoy style.